frontend/HelperFuncs/utils.js

```diff
@@ -475,6 +452,115 @@ export const findGenericGoals = async(transactionData) => {
+b. if differnce is or exceeds 5%, the goal amount is 5% 
+*/
+function findInvestmentGoals(transaction) {
+    if(transaction.difference < 0){
+        if(transaction.difference < -5){
+            // if difference exceeds 5%, goal amount is 5%
+            return({
+                'category': transaction.category[1],
+                'target': 5
+            })
+        } else{
+            // otherwise the value becomes the floor rounded difference since we're dealing with negative values
+            return({
+                'category': transaction.category[1],
+                'target': Math.floor(transaction.difference) * -1
+            })
+        }
+    }
+}
+
+
+/*
+helper function that tests if [transaction] meets criteria to be added to the
+database of goals when [transaction] is an expense, i.e. any category not investment_and_savings.
+An expense goal is added if the difference between a user's spending and the average
+among similar users is negative (i.e. this user is spending more than their peers)
+The 'target' or value users are encouraged to hit in a goal follows this rule:
+a. if difference is within 5%, the goal amount is the difference between user expenditure and average
+b. if differnce is or exceeds 5%, the goal amount is 5% 
+*/
+function findExpenseGoals(transaction) {
+    // logic for other categories is reversed from savings/investment
+    if(transaction.difference > 5){
+        return({
+            'category': transaction.category[1],
+            'target': 5
+        })
+    } else{
+        return({
+            'category': transaction.category[1],
+            'target': Math.ceil(transaction.difference)
+        })
+    }
+}
+
+
+/*
+helper that returns the list of available goals in the Goals database
+*/
+export const fetchAvailableGoals = async() => {
+    const response = await fetch(`http://localhost:3000/availablegoals`, 
+    { method: 'GET' })
+    if(!response.ok){
+        throw new Error('Network response was not ok at fetchAvailableGoals', Error);
+    }
+    const data = await response.json();
+    return data;
+}
+
+
+/*
+helper function to return strings that provide the user with concise details
+about the goal they're tracking / adding
+Example output: "lower [resturant] spending by [5]%"
+*/
+export function goalFormatter(category, target) {
+    let message = ''
+    // as with all other goal handling, the savings and investment goals are reversed, i.e. raise rather than lower
+    if(category === 'savings_account' || category === 'investment' || category === 'financial_planning'){
+        message += 'raise spending on '
+    } else {
+        message += 'lower spending on '
+    }
+    message += category + ' by ' + target.toString() + '%'
+    return message
+}
+
 
+/* 
+helper to append the requested goal onto the users Goals column and update
+the corresponding goal entry to include user as a tracked user
+*/
+export const followNewGoal = async(goalId, userId) => {
```

Comment:
is there a need for followNewGoal to be a helper? It's only used once in another file. I'm all for helper functions, but sometimes it's nicer to keep logic that relates to a certain feature in one file, especially if that logic isn't being used elsewhere.

Reply:
Yeah I don't see a use case for this endpoint outside the AddableGoal container so I can go ahead and move all endpoint logic in there if that would be cleaner

Reply:
refactored this and the sibling remove endpoint in 6714dd7